### PR TITLE
SCT-1147 add schema checker

### PR DESCRIPTION
**Pull request by MrCreosote**
Ensures the service won't start up with an incompatible DB schema.

Also makes index creation declarative via dicts.

### CODE REVIEW

**Comment by jayrbolton** [comment]
A couple minor comments, otherwise :+1:

File: src/jgikbase/idmapping/storage/mongo/id_mapping_mongo_storage.py [hunk 1]
```diff
@@ -48,11 +76,38 @@ def __init__(self, db: Database) -> None:
         not_none(db, 'db')
         self._db = db
         self._ensure_indexes()
+        self._check_schema()  # MUST happen after ensuring indexes
 
     def _ensure_indexes(self):
         try:
-            self._db[_COL_USERS].create_index(_FLD_USER, unique=True)
-            self._db[_COL_USERS].create_index(_FLD_TOKEN, unique=True)
+            for col in _INDEXES:
```
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/jgikbase/idmapping/storage/mongo/id_mapping_mongo_storage.py]: I've thought the same thing for the java version of this code, but never quite got around to it.

File: src/jgikbase/idmapping/storage/mongo/id_mapping_mongo_storage.py [hunk 2]
```diff
@@ -23,12 +22,41 @@
...
+
+# database collections
 _COL_USERS = 'users'
 
+# collection fields
 _FLD_AUTHSOURCE = 'auth'
 _FLD_USER = 'user'
 _FLD_TOKEN = 'hshtkn'
 
+_INDEXES = {_COL_USERS: [{'idx': _FLD_USER,
```
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/jgikbase/idmapping/storage/mongo/id_mapping_mongo_storage.py]: Yeah, something about that format sets my teeth on edge. Too much whitespace for my taste, I think.

[the remark this answers is not in the document]
Comment (jayrbolton) [reply src/jgikbase/idmapping/storage/mongo/id_mapping_mongo_storage.py]: Why does whitespace set your teeth on edge?

[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/jgikbase/idmapping/storage/mongo/id_mapping_mongo_storage.py]: *Too much* whitespace. Which is a matter of taste when it gets down to this kind of level.

**Pull request merged**